pkg/cmd/roachtest/run.go

```diff
@@ -209,6 +227,80 @@ func runTests(register func(registry.Registry), filter *registry.TestFilter) err
 	return err
 }
 
+// predictTestsToRun filters the tests to run based on certain criteria:
```

Comment:
Nit: I wouldn't put this information here. Instead, it can go in `testselector` (if you want it to be there). The reason is that as the selection criteria evolves, you are going to update that package, you are not necessarily going to update this one. From the perspective of this package, the only thing that matters is what the CSV says.

Reply:
moved the code to testselector as consumer